articles/why-you-should-use-celery-with-rabbitmq/index.md

```diff
@@ -0,0 +1,463 @@
+
+### What We Will Cover In This Article
+
+In this article, we’ll walk through what tasks queue are, see the examples of task queues (which includes celery), learn about message brokers. We will also look at why we need to use Celery and RabbitMQ while also seeing a basic tutorial that shows its installation and usage.
+
+### Understanding Celery & RabbitMQ
+
+Before going into why we should use Celery & RabbitMQ, let us understand what they are and what similar tools are available. Celery is classified as a *Task Queue* and RabbitMQ is classified as a *Message Broker*. Let's dive deeper into what these are.
+
+#### What is a Task Queue?
+
+A task queue is a data structure maintained by a job scheduler containing jobs to run.
+Task queue software also manages background work that must be executed outside of the usual HTTP request-response cycle. They are designed for asynchronous work.
+To further explain their usefulness, let’s say we have a web application that uses artificial intelligence to enhance images.
+As the number of users increase, the time to enhance an image drastically increases, which leads to a significant delay while enhancing.
+Hence, the need for a task queu/e software as it efficiently manages requests and ensures the application runs smoothly.
+
+#### Examples of Task Queue Software
+
+- Celery
+- Redis Queue
+- Taskmaster
+- Huey
+- tasq
+
+#### What is a Message Broker ?
+
+A message broker allows applications, systems and services to communicate and exchange information with each other. Now, with a task queue software in the background, the web app needs to know the status of the job in progress, plus any errors that might occur and the end result of the enhancement. A message broker dramatically eases this process since it’s built to make independent processes “talk to each other”
+
+#### Examples Of Message Brokers
+
+- IBM MQ
+- Beanstalk
+- RabbitMQ
+- Redis
+- Gearman
+- Solace
+
+### Why We Should Choose The Celery & RabbitMQ Combo
+
+Having understood what task queues are, let's look at celery. Celery is an open-source task queue software written in python. It’s incredibly lightweight, supports multiple brokers (RabbitMQ, Redis and Amazon SQS) and it also integrates with multiple web frameworks, e.g. Django, etc. Celery's asynchronous task queue allows execution of tasks and its concurrency makes it used in several production systems. For example, Instagram uses Celery to scale thousands of tasks to millions.
+
+However, task execution needs message brokers to work smoothly. Celery supports three message brokers as mentioned above. Although, for Amazon SQS, there is no support for remote monitoring.
+
+Redis as a message broker has its limitations, as doing replication is manual and requires additional work to implement it. Also, because Redis is an in-memory solution, if the machine runs out of memory when building queues up, there's a risk to lose tasks. Beanstalk also does not support replication.
```

Comment:
```suggestion
Redis, as a message broker has its limitations. Doing replication is manual and requires additional work to implement it. Also, because Redis is an in-memory solution, if the machine runs out of memory when building queues up, there's a risk to lose tasks. Beanstalk also does not support replication.
```